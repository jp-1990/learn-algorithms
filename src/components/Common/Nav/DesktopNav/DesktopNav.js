import React from "react"
import PropTypes from "prop-types"
import { Link } from "gatsby"

import styles from "./DesktopNav.module.scss"

const DesktopNav = ({ links }) => {
  // return jsx for given links
  const linkArray = links.map((e, i) => {
    return (
      <Link key={i} to={e.link}>
        {e.label}
      </Link>
    )
  })
  return (
    <nav className={styles.nav}>
      <div className={styles.logo}>LOGO</div>
      <div className={styles.links}>{linkArray}</div>
    </nav>
  )
}

DesktopNav.propTypes = {
  links: PropTypes.arrayOf(PropTypes.object),
}

export default DesktopNav
