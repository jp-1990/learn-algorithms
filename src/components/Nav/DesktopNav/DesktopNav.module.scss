.nav {
  width: 100%;
  height: 80px;
  background-color: $colorD;
  display: flex;
  align-items: center;
  justify-content: space-between;
  .logo {
    margin: 0 30px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 2rem;
    font-weight: 900;
    color: white;
  }
  .links {
    display: flex;
    a {
      text-decoration: none;
      color: white;
      font-weight: 300;
      font-size: 2rem;
      margin: 0 30px;
    }
  }
}
