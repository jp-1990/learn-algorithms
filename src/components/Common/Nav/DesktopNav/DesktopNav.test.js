import React from "react"
import renderer from "react-test-renderer"

import DesktopNav from "./DesktopNav"

describe("<DesktopNav />", () => {
  const defaultProps = [
    { label: "Data Structures", link: "/dataStructures" },
    { label: "Algorithms", link: "/algorithms" },
    { label: "About", link: "/about" },
  ]

  const wrapper = renderer
    .create(<DesktopNav links={[...defaultProps]} />)
    .toJSON()

  test("render", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
