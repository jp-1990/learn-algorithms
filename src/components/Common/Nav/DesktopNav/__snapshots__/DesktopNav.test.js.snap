// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DesktopNav /> render 1`] = `
<nav
  className="nav"
>
  <div
    className="logo"
  >
    LOGO
  </div>
  <div
    className="links"
  >
    <a
      href="/dataStructures"
    >
      Data Structures
    </a>
    <a
      href="/algorithms"
    >
      Algorithms
    </a>
    <a
      href="/about"
    >
      About
    </a>
  </div>
</nav>
`;
